orbit/cmd/desktop/desktop.go

```diff
@@ -102,23 +127,53 @@ func main() {
 			}()
 
 			return done
-		}()
+		}
 
+		// start a check as soon as the app starts
+		deviceEnabledChan := checkToken()
+
+		// this loop checks the `mtime` value of the token file and:
+		// 1. if the token file was modified, it disables the tray items until we
+		// verify the token is valid
+		// 2. calls (blocking) `checkToken` to verify the token is valid
 		go func() {
 			<-deviceEnabledChan
-			tic := time.NewTicker(5 * time.Minute)
+			tic := time.NewTicker(1 * time.Second)
 			defer tic.Stop()
 
 			for {
 				<-tic.C
+				expired, err := tokenReader.HasChanged()
+				switch {
+				case err != nil:
+					log.Error().Err(err).Msg("check token file")
+				case !expired:
+					continue
+				default:
+					log.Info().Msg("token file changed, rechecking")
+					disableTray()
+					<-checkToken()
+				}
+			}
+		}()
```

Comment:
_note: the git diff here is a bit misleading, this is just a new anonymous function._

@lucasmrod this is the only thing I had to change from the spec, I realized I needed to this late on Friday, so apologies for not checking first.

The spec says that Orbit shouldn't write the new token value to disk until it is ingested by Fleet, this causes Fleet Desktop to operate on an old token for a window of time: from the moment the token is ingested until Orbit can verify it was properly ingested and write the file.

On this implementation:

1. Orbit writes the token to disk as soon as it changes it
2. Fleet Desktop checks the `mtime` of the token file every 1 second (we could probably make this 5 seconds)
3. As soon as the token is changed, we disable all menu items and show "Connecting..." until we verify we have a valid token